### ISSUE TITLE
Add Round End gameplay

### ISSUE THREAD

**Issue by Peptide90** [Vault-Overseers/nuclear-14#511]
Rounds are currently lasting 4-5+ hours and this is far too long. Rounds should be topping out at 4 hours absolute max.

At the **3 hour** mark the round end countdown should be triggered. This allows for up to 30 minutes of events to slowly start happening to increasingly point towards a catastrophe. 

By around **3 hours 20**, players should get an overt announcement telling them of an impending issue. This may be accompanied by a hint of what the escape vehicle will be so people know where to head to.

**3 hours 25** - Escape vehicle arrives or is activated. You now have 5 minutes to get to it or seek appropriate shelter.

**3 hours 30** - Catastrophe happens. This can be a nuke detonating on the map, an earthquake or extreme radstorm.

There is then a period of roughly 10 minutes for people to survive in the aftermath of the event. Most people should die in this time if they didn't seek appropriate shelter. Round ends at 3 hours 40 unless extended for some reason.

**Comment by 128th** [comment]
I disagree. It should be a focus to encourage people to join the already ongoing round, even 2 - 3 hours in. There is no incentive for people to join current ongoing rounds, because most of the spots have been looted, trees and plants are cut, and the map is littered with guts and shit from decomposed bodies. It would be a good start to solve some of these problems first:
- Introduce replanting of rarely visited areas with plants every 1,5 - 2 hours.
- Restocking / recreating garbage piles with new stuff every 2 hours for late joiners.
- Redesigning decomposition system of corpses, when they do not permanently litter map with guts. Either make guts and viscera useful in some recipes (compost or food for example) or / and make them disappear after some time on their own. It's also possible to create spawn points for mobs to appear if the area is too littered with dead bodies / their guts.

I agree however with adding objectives to major faction, which completion would trigger round's end. But it should be something hard and elaborate, to prevent meta-rushing end of the round.

Also global events would be good on their own. Like each 2 - 3 hours something happening, which would change rest of the round or affect everybody, so they would be forced to adapt it for some time.
- Sandstorms
- Acid rains / ash snowstorms
- Mob migration
- Raider attack on different faction cites

**Comment by Peptide90** [comment]
All of these things are planned or semi implemented in some way. However our intent is not for peristent or extremely long form rounds. Rounds should still have an end so that a new round may start and players can reshuffle their roles.

Imagine you only get 1 day a week to play games and you really want to play as an NCR Officer but the round lasts for 8 hours so you don't get to. There is a place in this world for long form storytelling or persistence and we have discussed it previously, but for now a disaster to end the round and force people to react seems like a logical way to end a round. Maybe the times are off by an hour but still.

- Objectives are planned and maybe we can add certain objectives that can force a round end or "win" for a faction.
- Weather is semi implemented and will force people indoors depending on the event. I just need to add a system so it works at random.
- Plant respawning can easily be added.
- Loot piles that respawn are already in and are disabled because they aren't a good mechanic. We have a whole cargo system to implement through the caravan company to supplement this, as well as massive loot dungeons that will be very difficult if not impossible to solo as a means to add more loot.
- There are plans for an expedition system to smaller adjacent maps for loot at later times in a round.
- More crafting recipes for all mobs and mob parts are planned. Decomposition of what remains is easy to add.

Hope that clears up some of your concerns. It's all a WIP and unless someone actually codes in round end gameplay, we rely on admins or players to end rounds and that leads to incredibly long rounds.
